image-layout.md

```diff
@@ -115,4 +116,42 @@ $ cat ./blobs/sha256/e692418e4cbaf90ca69d05a66403747baa33ee08806650b51fab815ad7f
 [tar stream]
 ```
 
+## Index 
+
+Index is intended to be generated from the content in an image-layout after any updates have been made.
+Allows for mapping random access to the references in the image-layout, and needing only to read from the `oci-layout` once to build this map of references.
+
+**index** *array of objects*, OPTIONAL
```

Comment:
With a separate file, a long index only impacts ref-listing.  By putting it in `oci-layout`, a long index impacts *any* image-layout activity (because everyone should be checking to see “is this an image-layout version I understand” before trying to use the bits they need).

> Plus this truncated list you mention does not allow for having tagged metadata.

If annotating refs is important, I'd rather have `annotations` added to the descriptor model.  Do you have an example use case that shows that `annotations` are useful for refs but *not* for descriptors in general?